spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -84,7 +89,14 @@ public Resource resolve(String location, ResourceLoader resourceLoader) {
 							+ "so defaults are used.");
 				}
 			}
-			return new GoogleStorageResource(this.storage, location, this.settings.isAutoCreateFiles());
+
+			if (pathParts.length < 2) {
+				return new GoogleStorageResourceBucket(this.storage, bucketName,
+						this.settings.isAutoCreateFiles());
+			}
+
+			return new GoogleStorageResourceObject(this.storage, location,
```

Comment:
I'm guessing we're letting the GCS API error out for malformed GCS URIs, like `gs://bucket-name/asdasd/asdasd/asdasd`. I think I'm fine with that.

Reply:
Isn't `gs://bucket-name/asdasd/asdasd/asdasd` a valid path though?

Reply:
You're right -- nevermind that!